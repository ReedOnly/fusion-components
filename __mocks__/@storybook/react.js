import renderer from 'react-test-renderer';
import addons, { mockChannel } from '@storybook/addons';
import { withFusionContext } from '../../.storybook/withFusionContext';

addons.setChannel(mockChannel());

export const action = (actionName) => jest.fn();
const fusionContext = withFusionContext();
export const storiesOf = (groupName) => {
    const api = {
        decorators: [],
        parameters: {},

        add(storyName, story) {
            if (api.parameters.skipTest) {
                return;
            }

            describe(groupName, () => {
                it(storyName, () => {
                    // const decoratedStory = api.decorators.reduce((s, decorator) => {
                    //     try {
                    //         return decorator(s, { parameters: api.parameters });
                    //     } catch (e) {
                    //         return s;
                    //     }
                    // }, story);

                    const component = renderer.create(fusionContext(story));

                    // @TODO Enable custom names once released:
                    // > https://github.com/facebook/jest/pull/2094
                    expect(component.toJSON())
                        .toMatchSnapshot
                        // `${groupName}.${storyName}`
                        ();
                });
            });

            return api;
        },

        addDecorator() {
            return api;
        },

        addParameters(parameters) {
            api.parameters = { ...api.parameters, ...parameters };
            return api;
        },
    };

    return api;
};

export const addDecorator = () => {};
