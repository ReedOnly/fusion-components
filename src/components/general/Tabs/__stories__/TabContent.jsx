import React from "react";
import PropTypes from "prop-types";
import Button from "../../Button";
import styles from "./styles.less";

const firstChapterText = `Super mario landed gentry 
    erudite headmaster circus strongman, circus strongma
    n elit super mario got milk magnum pi dolor ipsum casual style
    landed gentry erudite headmaster barber shop quartet barber shop quartet, super mario groucho marx what a bounder magnum pi
    lemmy elit landed gentry lando calrissian got milk barber shop quartet circus strongman barber shop quartet mexican’t dolor ipsum casual 
    style erudite headmaster? Great dictator Daniel plainview theodore roosevelt ian botham clone zone shopper theodore roosevelt, pit fighter barber 
    shop quartet hello, we’re cockneys ian botham Daniel plainview che gu
    evara gent theodore roosevelt erudite headmaster great dictator clone zone shopper theodore roosevelt?`;

const secondChapterText = `Mouth coiffure brigadier great dictator. Sweat irrigator, dolor sit amet groucho-a-like grooming, sweat irrigator, 
    el snort basil fawlty Refined gentlemen rugged dolor sit amet groucho-a-like grooming frontiersman charlie chaplin graeme souness waiter?
    Consectetur erudite headmaster wario lando calrissian groucho marx. Will you do the fandango crumb catcher middle eastern despot louis xiii what a bounder?`;

const thirdChapterText = `Consectetur erudite headmaster wario lando calrissian groucho marx. Will you do the fandango 
    crumb catcher middle eastern despot louis xiii what a bounder?`;

export const FirstChapter = props => {
    return (
        <div className={styles.content}>
            <Button primary comfortable onClick={() => props.changeChapter("chapter2")}>
                Next Chapter
            </Button>
            <div className={styles.chapter}>{firstChapterText}</div>
        </div>
    );
};

export const SecondChapter = props => {
    return (
        <div className={styles.content}>
            <Button primary comfortable onClick={() => props.changeChapter("chapter1")}>
                Previous Chapter
            </Button>
            <Button primary comfortable onClick={() => props.changeChapter("chapter3")}>
                Next Chapter
            </Button>
            <div className={styles.chapter}>{secondChapterText}</div>
        </div>
    );
};

export const ThirdChapter = props => {
    return (
        <div className={styles.content}>
            <Button primary comfortable onClick={() => props.changeChapter("chapter2")}>
                Previous Chapter
            </Button>
            <div className={styles.chapter}>{thirdChapterText}</div>
        </div>
    );
};
FirstChapter.propTypes = {
    changeChapter: PropTypes.func.isRequired,
};

SecondChapter.propTypes = {
    changeChapter: PropTypes.func.isRequired,
};

ThirdChapter.propTypes = {
    changeChapter: PropTypes.func.isRequired,
};
