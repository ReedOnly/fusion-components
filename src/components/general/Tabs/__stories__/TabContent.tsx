import * as React from 'react';
import Button from '../../Button';
import styles from './styles.less';

type ItemProps = {
    changeItem: (Item: string) => void;
    prevItem?: string;
    nextItem?: string;
    content?: string;
};

export const Item: React.FC<ItemProps> = ({ changeItem, prevItem, nextItem, content }) => {
    const prevButton = prevItem ? (
        <Button primary comfortable onClick={() => prevItem && changeItem(prevItem)}>
            Previous Item
        </Button>
    ) : null;
    const nextButton = nextItem ? (
        <Button primary comfortable onClick={() => nextItem && changeItem(nextItem)}>
            Next Item
        </Button>
    ) : null;

    return (
        <div className={styles.content}>
            {prevButton}
            {nextButton}
            <div className={styles.item}>{content}</div>
        </div>
    );
};
