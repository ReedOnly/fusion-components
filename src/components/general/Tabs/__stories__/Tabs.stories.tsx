import * as React from "react";
import { storiesOf } from "@storybook/react";
import { Tabs, Tab } from "../index";
import { FirstChapter, SecondChapter, ThirdChapter } from "./TabContent";

const TabsStory = () => {
    const [activeTabKey, setActiveTabKey] = React.useState("chapter1");
    return (
        <Tabs activeTabKey={activeTabKey} onChange={tabKey => setActiveTabKey(tabKey)}>
            <Tab tabKey="chapter1" title="Chapter 1">
                <FirstChapter changeChapter={tabKey => setActiveTabKey(tabKey)} />
            </Tab>
            <Tab tabKey="chapter2" title="Chapter 2">
                <SecondChapter changeChapter={tabKey => setActiveTabKey(tabKey)} />
            </Tab>
            <Tab tabKey="chapter3" title="Chapter 3">
                <ThirdChapter changeChapter={tabKey => setActiveTabKey(tabKey)} />
            </Tab>
            <Tab tabKey="chapter4" title="Chapter 4" disabled />
        </Tabs>
    );
};
storiesOf("General components/Tabs", module)
    .addParameters({ jest: ["Tabs.stories.jsx"] })
    .add("Tabs", () => <TabsStory />);
