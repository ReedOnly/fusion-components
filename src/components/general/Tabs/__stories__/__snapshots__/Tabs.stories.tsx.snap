// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General|Tabs Default 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div>
      <div
        className=""
      >
        <div />
        <div />
        <a
          className=""
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          tabIndex={0}
        >
          <div
            className=""
          >
            Item 1
          </div>
        </a>
        <a
          className=""
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          tabIndex={0}
        >
          <div
            className=""
          >
            Item 2
          </div>
        </a>
        <a
          className=""
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          tabIndex={0}
        >
          <div
            className=""
          >
            Item 3
          </div>
        </a>
        <a
          className=""
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          tabIndex={0}
        >
          <div
            className=""
          >
            Item 4
          </div>
        </a>
        <a
          className=""
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          tabIndex={0}
        >
          <div
            className=""
          >
            Item 5 is very long
          </div>
        </a>
        <a
          className=""
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          tabIndex={0}
        >
          <div
            className=""
          >
            Item 6
          </div>
        </a>
        <span
          className=""
        >
          <div
            className=""
          >
            Item 7
          </div>
        </span>
      </div>
      <div>
        <div
          className="content"
        >
          <button
            className="container contained primary comfortable"
            disabled={false}
            onClick={[Function]}
            onClickCapture={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <span
              className="button"
            >
              Next Item
            </span>
          </button>
          <div
            className="item"
          >
            This is Item 1
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;
